pandas/_libs/interval.pyx

```diff
@@ -492,8 +492,9 @@ cdef class Interval(IntervalMixin):
     def __repr__(self) -> str:
 
         left, right = self._repr_base()
+        disp = str if isinstance(left, np.generic) else repr
         name = type(self).__name__
-        repr_str = f"{name}({repr(left)}, {repr(right)}, closed={repr(self.closed)})"
+        repr_str = f"{name}({disp(left)}, {disp(right)}, closed={repr(self.closed)})"
```

Comment:
maybe the thing to do is unpack numpy scalars at an earlier stage?

Reply:
But we should still store numpy scalars for left/right on Interval right?

Reply:
i was thinking store native scalars

Reply:
Hmm might make sense, not sure if there's any larger ramifications of that chance. Might be better to experiment in a follow up?

Reply:
sure